### PR TITLE
Always show RemoteSocket.Exists in json

### PR DESCRIPTION
**Pull request by dcermak**
The `Exists` field of the `RemoteSocket` struct is marshaled to json with the `omitempty` setting. This has the disadvantage that by default `podman info` shows a `remotePath` entry (the remote path is set in `pkg/domain/infra/abi/systems.go`: `(*ContainerEngine).Info`) but not that this path does not exist:
```
❯ podman info --format json | jq .host.remoteSocket
{
  "path": "/run/user/1000/podman/podman.sock"
}
```

By removing the `omitempty`, we ensure that the existence is always shown:
```
❯ bin/podman info --format json | jq .host.remoteSocket
{
  "path": "/run/user/1000/podman/podman.sock",
  "exists": false
}
```

#### Does this PR introduce a user-facing change?

```release-note
- always display the existence of the podman socket in `podman info`
```

### CODE REVIEW

**Review by ricardobranco777 — COMMENTED** [review]
Nice!

**Review by vrothberg — COMMENTED** [review]
Thanks!  Please update the tests to make CI pass.

**Comment by rhatdan** [comment]
/approve
LGTM

**Comment by dcermak** [comment]
Any idea what the issue with the CI is?

**Comment by vrothberg** [comment]
> Any idea what the issue with the CI is?

Looks like a flake.  I'll refired and will check in case it fails again.

**Comment by Luap99** [comment]
This is not flake, the exact check which you added is failing so this is cannot be a flake and even if it were one we should never merge a flake added in a new test and rather have them fixed before merging.

The problem here is likely that on debian the podman.socket is enabled so the remote socket will always show as active

**Comment by vrothberg** [comment]
> This is not flake, the exact check which you added is failing so this is cannot be a flake and even if it were one we should never merge a flake added in a new test and rather have them fixed before merging.

Nobody suggested that.  I didn't take a look at the logs.
 
> The problem here is likely that on debian the podman.socket is enabled so the remote socket will always show as active

Nice catch!

**Comment by Luap99** [comment]
Right but blindly rerunning tests without looking at the logs is a bad habit, we should not do that.
It will result in us merging new flakes, if the test which is modified in a PR fails that should always mean fix it and not waste time rerunning.

**Comment by Luap99** [comment]
Confirmed running hack/get_ci_vm.sh shows me that the socket is enabled,
so we should disabled it in the CI setup (look in `contrib/cirrus/` somwhere). @cevich would know where the best place is?
We need to run `systemctl disable --now podman.socket` and I guess it would make the most sense to that that in the image building process.

**Comment by cevich** [comment]
> the socket is enabled

Hmmmmm, I don't think that's the default when simply installing the RPM or `make install`...yeah, in fact [we disable it when replacing the RPM](https://github.com/containers/podman/blob/cd5ce637249e58298a2d55ec54bd2b77b6274b4d/contrib/cirrus/lib.sh#L272) at CI setup time.  Oh! wait, maybe `io.podman.socket` name is wrong.  That's probably what needs fixing.

**Comment by Luap99** [comment]
we are talking about debian only here, fedora seems to be disabled already

**Comment by cevich** [comment]
> we are talking about debian only here

That's interesting, I'm pretty sure that same `remove_packaged_podman_files()` is also used for Debian.  That function hasn't been touched in ages, maybe the service name is different on Debian?

**Comment by cevich** [comment]
Investigating a fix...

**Comment by Luap99** [comment]
socket and service are called `podman.socket` and `podman.service` for a long time now.

**Comment by cevich** [comment]
...confirmed, on Debian the service is called 'podman.socket'.  The fix is simple:

```diff
diff --git a/contrib/cirrus/lib.sh b/contrib/cirrus/lib.sh
index 265763b8b..b48b46f1c 100644
--- a/contrib/cirrus/lib.sh
+++ b/contrib/cirrus/lib.sh
@@ -269,7 +269,7 @@ remove_packaged_podman_files() {
     req_env_vars OS_RELEASE_ID

     # If any binaries are resident they could cause unexpected pollution
-    for unit in io.podman.service io.podman.socket
+    for unit in io.podman.service io.podman.socket podman.socket podman-auto-update.timer
     do
         for state in enabled active
         do
```

**Comment by Luap99** [comment]
not on debian only, everywhere! There is no io.podman.service and io.podman.socket for several years now.

**Comment by dcermak** [comment]
I've included the suggested fix from @cevich. Sadly, all CI jobs on cirrus are now failing to start.

**Comment by vrothberg** [comment]
Restarted.  Some infra fart.

**Review by vrothberg — APPROVED** [review]
LGTM

**Comment by Luap99** [comment]
This is still failing likely because stopping a socket unit will not clean up the socket file in the file system.
I think these units must be disabled at image built time so the socket is never created when the VM boots up.

But anyway we do not need to block this for now. You can add `rm -f /run/podman/podman.sock` after the units are disabled then I am sure it should work fine.

**Comment by cevich** [comment]
Thanks for the suggestions Paul.  It won't hurt to have both.  Adding that rm into `remove_packaged_podman_files()` will cover all distros in all cases.  I'll open an issue to remove the socket in the images.

File: contrib/cirrus/lib.sh
Comment (cevich) [inline contrib/cirrus/lib.sh:296]: Perfect!

File: contrib/cirrus/lib.sh
Comment (cevich) [inline contrib/cirrus/lib.sh:272]: Are you sure the `io.podman.service` and `io.podman.socket` names aren't used in any CI distro.?  IIUC this function should gracefully handle invalid names for a distro. even if they're valid for another.

Comment (cevich) [reply contrib/cirrus/lib.sh:272]: Historical note: It's entirely possible these names came from Ubuntu, a very, very, very log time ago.

Comment (Luap99) [reply contrib/cirrus/lib.sh:272]: We test fedora and debian, neither of those uses them. And no distro will use them because they all use upstream defaults (contrib/systemd/system), they would explicitly rename untis. These are legacy unit names for the old varlink socket.

Comment (cevich) [reply contrib/cirrus/lib.sh:272]: Ooooohhhh damn, the bloody varlink bites again! Lol, thanks for the clarification Paul :+1:

**Review by cevich — APPROVED** [review]
The CI changes LGTM (in case that wasn't obvious).

**Comment by rhatdan** [comment]
/lgtm
/hold cancel
Thanks @dcermak

**Pull request merged**
